### PR TITLE
changes openssh addr validation to allow hosts

### LINKED ISSUE #25935: Address for openssh node def not allowed to put dns name
Expected behavior:

DNS and IPs are valid for addresses for Node entries. In many cases the OpenSSH server is reachable by a DNS name, not just a fixed IP.  Also alternate IPs allows for dns names while this does not.

Current behavior:

Only IPs are allowed in the `addr` spec while you are allowed to put DNS entries in the 

Bug details:
- Teleport version: 13.0.0
- Recreation steps

Attempt to set the `addr` as something besides a IP

```yaml
kind: node
version: v2
sub_kind: openssh
metadata:
  name: ubuntu-openssh
  labels:
    tier: dev 
spec:
  addr: ubuntuhost:22
  hostname: openssh-node
```

```bash
tctl create -f openssh.yaml 
ERROR: invalid Addr "ubuntuhost:22": ParseAddr("ubuntuhost"): unexpected character (at "ubuntuhost")
```

### PR DESCRIPTION
**Pull request by stevenGravy**
fixes #25935

### CODE REVIEW

**Review by capnspacehook — APPROVED** [review]

**Review by rosstimothy — COMMENTED** [review]
Can we some test coverage?

**Review by r0mant — COMMENTED** [review]
@capnspacehook @stevenGravy Does everything work as expected when using hostname in agentless mode? Have you verified this?

File: api/types/server_test.go [hunk 1]
```diff
@@ -211,6 +211,39 @@ func TestServerCheckAndSetDefaults(t *testing.T) {
...
+						Namespace: defaults.Namespace,
+					},
+					Spec: ServerSpecV2{
+						Addr:     "example:22",
+						Hostname: "openssh-node",
+					},
+				}
+				require.Equal(t, expectedServer, s)
+			},
+    },
```
[the remark this answers is not in the document]
Comment (stevenGravy) [reply api/types/server_test.go:246]: Yes, I have confirmed it.

[the remark this answers is not in the document]
Comment (stevenGravy) [reply api/types/server_test.go:246]: working on lint fixes

File: api/types/server_test.go [hunk 2]
```diff
@@ -211,6 +211,39 @@ func TestServerCheckAndSetDefaults(t *testing.T) {
...
+            Namespace: defaults.Namespace,
+          },
+          Spec: ServerSpecV2{
+            Addr:     "example:22",
+            Hostname: "openssh-node",
+          },
+        }
+        require.Equal(t, expectedServer, s)
+      },
+    },
```
[the remark this answers is not in the document]
Comment (stevenGravy) [reply api/types/server_test.go]: thanks @rosstimothy , did not copy cleanly, fixed

**Comment by stevenGravy** [comment]
> Can we some test coverage?

I have added a node configuration in the server test that includes a dns address with a port.

**Comment by stevenGravy** [comment]
@rosstimothy @r0mant thanks for the reviews. We should be good to go now with tests impelmented, lint fixes. Otherwise please let me know anything else that needs addressing.

**Review by r0mant — APPROVED** [review]

**Pull request merged**